libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -491,17 +491,27 @@ bool AP_Arming::gps_checks(bool report)
             }
         }
 
-        //GPS OK?
-        if (!AP::ahrs().home_is_set() ||
-            gps.status() < AP_GPS::GPS_OK_FIX_3D) {
-            check_failed(ARMING_CHECK_GPS, report, "Bad GPS Position");
-            return false;
-        }
+        for (uint8_t i = 0; i < gps.num_sensors(); i++) {
+            if (gps.get_type(i) == AP_GPS::GPS_Type::GPS_TYPE_NONE) {
+                if (gps.primary_sensor() == i) {
+                    check_failed(ARMING_CHECK_GPS, report, "GPS primary not configured");
+                    return false;
+                }
+                continue;
+            }
 
-        //GPS update rate acceptable
-        if (!gps.is_healthy()) {
-            check_failed(ARMING_CHECK_GPS, report, "GPS is not healthy");
-            return false;
+            //GPS OK?
+            if (!AP::ahrs().home_is_set() ||
+                gps.status(i) < AP_GPS::GPS_OK_FIX_3D) {
+                check_failed(ARMING_CHECK_GPS, report, "GPS %i: Bad Position", i+1);
```

Comment:
```suggestion
                check_failed(ARMING_CHECK_GPS, report, "GPS %i: Bad fix", i+1);
```

This may not be a better suggestion, we really aren't checking the position, but fix type. But I'm not to concerned here.